### PR TITLE
sed: BSD compatibility

### PR DESCRIPTION
**Pull request by dannywillems**
sed on Mac OS needs the extension with the option -i.
I tried on a Debian and it's also working.

### CODE REVIEW

**Comment by vasilisp** [comment]
I don't understand how this could be working on Debian, are you sure? `.css` is parsed as an argument detached from `-i`, so it becomes the `sed` expression, and it happens to be invalid.

```
sed: -e expression #1, char 1: unknown command: `.'
```

File: template.distillery/Makefile.style [hunk 1]
```diff
@@ -27,7 +27,7 @@ $(CSSDIR):
 $(DEST): $(LOCAL_CSS) | $(CSSDIR)
 	HASH=`cat $< | md5sum | cut -d ' ' -f 1` && \
 	cp $< $(CSS_PREFIX)_$$HASH.css && \
-	sed -i '1s/^/@charset "UTF-8";/' $(CSS_PREFIX)_$$HASH.css && \
+	sed -i .css '1s/^/@charset "UTF-8";/' $(CSS_PREFIX)_$$HASH.css && \
```
Comment (pwbs) [inline template.distillery/Makefile.style]: simplest way is probably to write `sed -i -e '1s/^/@charset "UTF-8";/' $(CSS_PREFIX)_$$HASH.css && \ `

**Comment by pwbs** [comment]
@dannywillems It's more BSD compatibility than mac OS compatibility... ;)

File: template.distillery/Makefile.style (hunk 1 above)
Comment (dannywillems) [inline template.distillery/Makefile.style]: @vasilisp my bad, I was on the wrong session when I ~~tried~~ thought I tried on Debian --'.

**Comment by dannywillems** [comment]
@vasilisp @pwbs -e works.

**Comment by pwbs** [comment]
normally 
`echo pppppPPP > d ;  sed -i.old -e 's/p/P/g' d`
gives the same results with BSD sed and GNU sed. So, perhaps use this?

**Comment by vasilisp** [comment]
@dannywillems , don't you end up with an auxiliary file with the `-e` suffix? See discussion in https://github.com/ocsigen/eliom/commit/54240d4ca2f617206b8158615ec42dcf104e3667 .

**Comment by pwbs** [comment]
otherwise, don't use -i (but use a temporary file) or detect version of sed as in
```bash
if (( sed --version|grep -q GNU )); then
SEDI='sed -i '
else
SEDI='sed -i ""'
fi
# and then use $SEDI ...
```

**Comment by dannywillems** [comment]
@pwbs seems to have no --version on BSD. So need to detect the error.

```
sed --version
sed: illegal option -- -
```

**Comment by pwbs** [comment]
just send error to void: `2> /dev/null`

**Pull request closed without merging**